parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -487,6 +487,33 @@ configAddons() {
 }
 {{end}}
 {{- if HasNSeriesSKU}}
+
+installGPUDriversRun() {
+    {{- /* there is no file under the module folder, the installation failed, so clean up the dirty directory
+    when you upgrade the GPU driver version, please help check whether the retry installation issue is gone,
+    if yes please help remove the clean up logic here too */}}
+    set -x
+    MODULE_NAME="nvidia"
+    NVIDIA_DKMS_DIR="/var/lib/dkms/${MODULE_NAME}/${GPU_DV}"
+    KERNEL_NAME=$(uname -r)
+    if [ -d "${NVIDIA_DKMS_DIR}" ]; then
+        echo "the dkms module folder exists, we need to do the clean up first before retry to install."
+        if [ -x "$(command -v dkms)" ]; then
+          dkms remove -m ${MODULE_NAME} -v ${GPU_DV} -k ${KERNEL_NAME}
+        else
+          echo "the dkms tool is not found, so we just remove the directory to have a try"
```

Comment:
I don't think we can tolerate dkms not being present on the system. The nvidia modproe service needs it:

https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/nodecustomdata.yml#L273

If there is evidence of dkms not being present on the system, we want to fix that elsewhere (such as when it is installed via apt-get install)

Reply:
the "dkms remove" here is just to do the clean up before we retry the installation.

Reply:
and then the nvidia-drivers-$GPU_DV will install that for us.